lib/utils.bash

```diff
@@ -174,6 +176,10 @@ get_version_in_dir() {
   done
 }
 
+version_file_name() {
+  printf "%s" "${ASDF_DEFAULT_TOOL_VERSIONS_FILENAME:-.tool-versions}"
+}
```

Comment:
This helper function encapsulates the version file logic and prints the correct version file name.

Note that this has nothing to do with legacy version files. This is only for asdf version files (default of `.tool-versions` of course).